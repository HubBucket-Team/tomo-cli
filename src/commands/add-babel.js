import {
    allDoNotExist,
    BabelConfigModuleEditor,
    install,
    someDoExist
} from '../utils';

const cfg = new BabelConfigModuleEditor();

const BABEL_DEPENDENCIES = [
    '@babel/cli',
    '@babel/core',
    '@babel/runtime'
];
const BABEL_PRESETS = [
    '@babel/preset-env'
];
const BABEL_PLUGINS = [
    '@babel/plugin-transform-runtime',
    '@babel/plugin-proposal-class-properties',
    '@babel/plugin-proposal-export-default-from',
    '@babel/plugin-proposal-optional-chaining'
];
const BABEL_REACT_PRESET = [
    '@babel/preset-react'
];
export default [
    {
        text: 'Create Babel config file',
        task: () => cfg.create(),
        condition: () => allDoNotExist('babel.config.js', '.babelrc', '.babelrc.js')
    },
    {
        text: 'Install Babel dependencies',
        task: ({skipInstall}) => install(BABEL_DEPENDENCIES, {dev: true, skipInstall}),
        condition: () => someDoExist('package.json')
    },
    {
        text: 'Install Babel presets',
        task: ({skipInstall}) => install(BABEL_PRESETS, {dev: true, skipInstall}),
        condition: () => someDoExist('package.json')
    },
    {
        text: 'Install Babel plugins',
        task: ({skipInstall}) => install(BABEL_PLUGINS, {dev: true, skipInstall}),
        condition: () => someDoExist('package.json')
    },
    {
        text: 'Install Babel React preset',
        task: ({skipInstall}) => install(BABEL_REACT_PRESET, {dev: true, skipInstall}),
        condition: ({useReact}) => (useReact && someDoExist('package.json')),
        optional: ({useReact}) => useReact
    },
    {
        text: 'Add React support to Babel configuration file',
        task: () => cfg.extend({
            presets: [...BABEL_PRESETS, ...BABEL_REACT_PRESET]
        }),
        condition: ({useReact}) => (useReact && someDoExist('babel.config.js')),
        optional: ({useReact}) => useReact
    }
];