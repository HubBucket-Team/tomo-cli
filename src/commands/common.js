import {mkdirp} from 'fs-extra';
import {
    allDoNotExist,
    PackageJsonEditor
} from '../utils';

export const createPackageJson = [
    {
        text: 'Create package.json',
        task: () => {
            const pkg = new PackageJsonEditor();
            pkg.create();
        },
        condition: () => allDoNotExist('package.json')
    }
];
export const createSourceDirectory = [
    {
        text: 'Create source directory',
        task: ({sourceDirectory}) => mkdirp(sourceDirectory),
        condition: ({sourceDirectory}) => allDoNotExist(sourceDirectory)
    }
];