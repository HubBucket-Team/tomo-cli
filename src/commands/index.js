/* eslint-disable no-magic-numbers */
import {
    createPackageJson,
    createSourceDirectory
} from './common';
import addA11y from './add-a11y';
import addBabel from './add-babel';
import addEsdoc from './add-esdoc';
import addEslint from './add-eslint';
import addJest from './add-jest';
import addMarionette from './add-marionette';
import addPostcss from './add-postcss';
import addWebpack from './add-webpack';

const createProject = [
    ...createPackageJson,
    ...createSourceDirectory,
    ...addBabel,
    ...addEslint,
    ...addJest
];
const create = {
    project: createProject,
    app: [
        ...createProject,
        ...addMarionette
    ],
    server: [
        ...createProject
    ]
};
const add = {
    a11y: addA11y,
    babel: addBabel,
    docs: addEsdoc,
    eslint: [
        ...addBabel,
        ...addEslint
    ],
    jest: [
        ...addBabel,
        ...addJest
    ],
    postcss: addPostcss,
    webpack: addWebpack
};

module.exports = {
    add,
    create,
    new: create// alias for create
};