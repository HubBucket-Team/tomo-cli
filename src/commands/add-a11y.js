import {
    PackageJsonEditor,
    install,
    uninstall
} from '../utils';
import {allDoExist} from '../utils/common';
/**
 * @type {task[]}
 * @see http://pa11y.org/
 */
export const addA11y = [
    {
        text: 'Add pa11y tasks to package.json',
        task: async ({outputDirectory}) => {
            const scripts = {
                'lint:a11y': `pa11y ${outputDirectory}/index.html`
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Install pa11y',
        task: ({skipInstall}) => install(['pa11y'], {dev: true, skipInstall}),
        condition: ({isNotOffline}) => isNotOffline && allDoExist('package.json')
    }
];
export const removeA11y = [
    {
        text: 'Remove pa11y tasks from package.json',
        task: async () => {
            const scripts = {
                'lint:a11y': undefined
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Uninstall pa11y',
        task: () => uninstall(['pa11y']),
        condition: ({skipInstall}) => !skipInstall && allDoExist('package.json') && (new PackageJsonEditor()).hasAll('pa11y'),
        optional: ({skipInstall}) => !skipInstall
    }
];
export default addA11y;