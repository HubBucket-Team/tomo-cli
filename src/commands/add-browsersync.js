import {PackageJsonEditor, install} from '../utils';
import {allDoExist, someDoExist} from '../utils/common';

const BROWSERSYNC_DEPENDENCIES = [
    'browser-sync',
    'npm-run-all'
];
/**
 * @type {task[]}
 * @see https://www.browsersync.io/docs/command-line
 */
export const addBrowsersync = [
    {
        text: 'Add Browsersync tasks to package.json',
        task: async ({outputDirectory}) => {
            const scripts = {
                prestart: 'npm run build',
                start: 'npm-run-all --parallel build:watch build:css:watch serve:dev',
                'serve:dev': `browser-sync start --server ${outputDirectory} --files ${outputDirectory}`
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json', 'webpack.config.js', 'postcss.config.js')
    },
    {
        text: 'Install Browsersync dependencies',
        task: ({skipInstall}) => install(BROWSERSYNC_DEPENDENCIES, {dev: true, skipInstall}),
        condition: ({isNotOffline}) => isNotOffline && someDoExist('package.json')
    }
];