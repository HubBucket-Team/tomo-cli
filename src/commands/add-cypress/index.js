import {join} from 'path';
import {EslintConfigModuleEditor, PackageJsonEditor, install, uninstall} from '../../utils';
import {allDoExist, allDoNotExist} from '../../utils/common';
import {Scaffolder} from '../../utils/Scaffolder';
import createJsonEditor from '../../utils/createJsonEditor';

const ALWAYS = () => true;
const CYPRESS_DEPENDENCIES = [
    'cypress',
    'cypress-match-screenshot'
];
/**
 * @type {task[]}
 * @see https://www.cypress.io/
 */
export const addCypress = [
    {
        text: 'Create Cypress config file',
        task: async ({port}) => {
            const Editor = createJsonEditor('cypress.json', {
                baseUrl: `http://localhost:${port}`
            });
            await (new Editor())
                .create()
                .commit();
        },
        condition: () => allDoNotExist('cypress.json')
    },
    {
        text: 'Add Cypress test tasks to package.json',
        task: async () => {
            const scripts = {
                cypress: 'cypress open',
                'test:e2e': 'npm-run-all --parallel start cypress'
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Add Cypress global variable to .eslintrc.js',
        task: async () => {
            const globals = {cy: true};
            await (new EslintConfigModuleEditor())
                .extend({globals})
                .commit();
        },
        condition: () => allDoExist('.eslintrc.js')
    },
    {
        text: 'Copy Cypress files',
        task: async ({overwrite}) => {
            await (new Scaffolder(join(__dirname, 'templates')))
                .overwrite(overwrite)
                .target('cypress/support')
                .copy('support/index.js', 'index.js')
                .copy('support/commands.js', 'commands.js')
                .target('cypress/integration')
                .copy('visual-regression.test.js')
                .commit();
        },
        condition: ALWAYS
    },
    {
        text: 'Install Cypress dependencies',
        task: ({skipInstall}) => install([...CYPRESS_DEPENDENCIES, 'npm-run-all'], {dev: true, skipInstall}),
        condition: ({isNotOffline}) => isNotOffline && allDoExist('package.json')
    }
];
export const removeCypress = [
    {
        text: 'Delete Cypress config file',
        task: async () => {
            const Editor = createJsonEditor('cypress.json');
            await (new Editor())
                .delete()
                .commit();
        },
        condition: () => allDoExist('cypress.json')
    },
    {
        text: 'Remove Cypress test tasks from package.json',
        task: async () => {
            const scripts = {
                cypress: undefined,
                'test:e2e': undefined
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Remove Cypress global variable from .eslintrc.js',
        task: async () => {
            const globals = {cy: false};
            await (new EslintConfigModuleEditor())
                .extend({globals})
                .commit();
        },
        condition: () => allDoExist('.eslintrc.js')
    },
    {
        text: 'Uninstall Cypress dependencies',
        task: () => uninstall(CYPRESS_DEPENDENCIES),
        condition: ({skipInstall}) => !skipInstall && allDoExist('package.json') && (new PackageJsonEditor()).hasAll(...CYPRESS_DEPENDENCIES),
        optional: ({skipInstall}) => !skipInstall
    }
];
export default addCypress;