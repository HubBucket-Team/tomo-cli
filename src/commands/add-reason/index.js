import {join} from 'path';
import {BsConfigJsonEditor, PackageJsonEditor, install, uninstall} from '../../utils';
import {allDoExist, allDoExistSync, allDoNotExist, allDoNotExistSync} from '../../utils/common';
import {Scaffolder} from '../../utils/Scaffolder';

const DEPENDENCIES = [
    'reason-react'
];
const DEV_DEPENDENCIES = [
    'bs-platform'
];
/**
 * @type {task[]}
 * @see https://reasonml.github.io/reason-react/en/
 */
export const addReason = [
    {
        text: 'Create bsconfig.json file',
        task: async () => {
            const {name} = (new PackageJsonEditor()).read();
            await (new BsConfigJsonEditor())
                .create()
                .extend({name})
                .commit();
        },
        condition: () => allDoNotExist('bsconfig.json')
    },
    {
        text: 'Add Reason scripts to package.json',
        task: async () => {
            const scripts = {
                'build:reason': 'bsb -make-world -clean-world',
                'watch:reason': 'npm run build:reason -- -w'
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Install ReasonReact dependencies',
        task: async ({skipInstall}) => {
            await install(DEPENDENCIES, {skipInstall});
            await install(DEV_DEPENDENCIES, {dev: true, skipInstall});
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Copy ReasonReact boilerplate files',
        task: async ({overwrite, sourceDirectory}) => {
            await (new Scaffolder(join(__dirname, 'templates')))
                .overwrite(overwrite)
                .target(`${sourceDirectory}/components`)
                .copy('App.re')
                .copy('Example.re')
                .commit();
        },
        condition: ({useReact}) => useReact,
        optional: ({useReact}) => useReact
    },
    {
        text: 'Configure Webpack Reason support',
        task: async () => {

        },
        condition: () => allDoExist('webpack.config.js'),
        optional: () => allDoExistSync('webpack.config.js')
    },
    {
        text: 'Configure Parcel.js Reason support',
        task: async () => {

        },
        condition: ({useParcel}) => useParcel || allDoNotExist('webpack.config.js', 'rollup.config.js'),
        optional: ({useParcel}) => useParcel || allDoNotExistSync('webpack.config.js', 'rollup.config.js')
    },
    {
        text: 'Configure Rollup.js Reason support',
        task: async () => {

        },
        condition: ({useRollup}) => useRollup || allDoExist('rollup.config.js'),
        optional: ({useRollup}) => useRollup || allDoExistSync('rollup.config.js')
    }
];
export const removeReason = [
    {
        text: 'Remove Reason scripts from package.json',
        task: async () => {
            const scripts = {
                'build:reason': undefined,
                'watch:reason': undefined
            };
            await (new PackageJsonEditor())
                .extend({scripts})
                .commit();
        },
        condition: () => allDoExist('package.json')
    },
    {
        text: 'Uninstall Reason dependencies',
        task: () => uninstall([...DEPENDENCIES, ...DEV_DEPENDENCIES]),
        condition: ({skipInstall}) => !skipInstall && allDoExist('package.json') && (new PackageJsonEditor()).hasAll(...DEPENDENCIES, ...DEV_DEPENDENCIES) // eslint-disable-line max-len
    }
];
export default addReason;