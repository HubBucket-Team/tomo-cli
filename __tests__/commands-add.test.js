import {
    fileContents,
    getDirectoryTree,
    run,
    useTemporaryDirectory
} from './tomo-test';
import {createPackageJson} from '../src/commands/common';
import addEsdoc from '../src/commands/add-esdoc';
import addEslint from '../src/commands/add-eslint';
// import addRust from '../src/commands/add-rust';
import addMarionette from '../src/commands/add-marionette';
import {addReason} from '../src/commands/add-reason';

jest.mock('is-online', () => (async () => true));

describe('"Add" commands', () => {
    let tempDirectory;
    const skipInstall = true;
    const useReact = true;
    const [setTempDir, cleanupTempDir] = useTemporaryDirectory();
    beforeEach(async () => {
        tempDirectory = await setTempDir();
        process.chdir(tempDirectory);
    });
    afterEach(async () => {
        await cleanupTempDir();
    });
    test('add-esdoc', async () => {
        const sourceDirectory = 'src';
        const options = {skipInstall, sourceDirectory};
        await run(createPackageJson, {});
        const pre = fileContents('./package.json');
        expect(pre).toMatchSnapshot();
        await run(addEsdoc, options);
        const tree = getDirectoryTree(tempDirectory);
        expect(tree).toMatchSnapshot();
        const post = fileContents('./package.json');
        expect(post).toMatchSnapshot();
        const contents = fileContents('./esdoc.conf.json');
        expect(contents).toMatchSnapshot();
    });
    test('add-esdoc (with React)', async () => {
        const sourceDirectory = 'src';
        const options = {skipInstall, sourceDirectory, useReact};
        await run(addEsdoc, options);
        const tree = getDirectoryTree(tempDirectory);
        expect(tree).toMatchSnapshot();
        const contents = fileContents('./esdoc.conf.json');
        expect(contents).toMatchSnapshot();
    });
    test('add-eslint', async () => {
        const sourceDirectory = 'src';
        const options = {skipInstall, sourceDirectory};
        await run(addEslint, options);
        const tree = getDirectoryTree(tempDirectory);
        expect(tree).toMatchSnapshot();
        const contents = fileContents('./.eslintrc.js');
        expect(contents).toMatchSnapshot();
    });
    test('add-eslint (with React)', async () => {
        const reactVersion = '16.2';
        const sourceDirectory = './src';
        const options = {reactVersion, skipInstall, sourceDirectory, useReact};
        await run(addEslint, options);
        const tree = getDirectoryTree(tempDirectory);
        expect(tree).toMatchSnapshot();
        const contents = fileContents('./.eslintrc.js');
        expect(contents).toMatchSnapshot();
    });
    test('add-marionette', async () => {
        const sourceDirectory = './src';
        const options = {skipInstall, sourceDirectory};
        await run(addMarionette, options);
        const tree = getDirectoryTree(tempDirectory);
        expect(tree).toMatchSnapshot();
    });
});
describe('Add Reason', () => {
    let tempDirectory;
    const outputDirectory = './dist';
    const sourceDirectory = 'src';
    const useReact = true;
    const skipInstall = true;
    const [setTempDir, cleanupTempDir] = useTemporaryDirectory();
    beforeEach(async () => {
        tempDirectory = await setTempDir();
        process.chdir(tempDirectory);
    });
    afterEach(async () => {
        await cleanupTempDir();
    });
    test('with Webpack', async () => {
        const options = {outputDirectory, skipInstall, sourceDirectory, useReact};
        await run(createPackageJson, {});
        await run(addReason, options);
        const pkg = fileContents('package.json');
        const cfg = fileContents('bsconfig.json');
        const tree = getDirectoryTree(tempDirectory);
        expect(pkg).toMatchSnapshot();
        expect(cfg).toMatchSnapshot();
        expect(tree).toMatchSnapshot();
    });
});