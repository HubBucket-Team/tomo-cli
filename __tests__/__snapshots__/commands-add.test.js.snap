// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"Add" commands add-electron --use-parcel 1`] = `
"{
    \\"name\\": \\"tomo-native-app\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"Native Desktop application built with Electron\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"main\\": \\"index.js\\",
    \\"scripts\\": {
        \\"build:electron\\": \\"npm-run-all build:es build:css\\",
        \\"prestart:electron\\": \\"npm run build:electron\\",
        \\"start:electron\\": \\"electron index --enable-logging\\",
        \\"dev\\": \\"npm-run-all --parallel watch:es start:electron\\"
    }
}
"
`;

exports[`"Add" commands add-electron 1`] = `
"{
    \\"name\\": \\"tomo-native-app\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"Native Desktop application built with Electron\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"main\\": \\"index.js\\",
    \\"scripts\\": {
        \\"build:electron\\": \\"npm-run-all build:es build:css\\",
        \\"prestart:electron\\": \\"npm run build:electron\\",
        \\"start:electron\\": \\"electron index --enable-logging\\",
        \\"dev\\": \\"npm-run-all --parallel watch:es watch:css start:electron\\"
    }
}
"
`;

exports[`"Add" commands add-electron 2`] = `
"{
    name: tomo-test,
    children: [
        {
            name: bin,
            children: [
                {
                    name: preload.js,
                    size: 314,
                    type: file
                }
            ],
            size: 314,
            type: directory
        },
        {
            name: index.js,
            size: 2194,
            type: file
        },
        {
            name: package.json,
            size: 467,
            type: file
        }
    ],
    size: 2975,
    type: directory
}
"
`;

exports[`"Add" commands add-esdoc (with React) 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: esdoc.conf.json,
            size: 417,
            type: file
        }
    ],
    size: 417,
    type: directory
}
"
`;

exports[`"Add" commands add-esdoc (with React) 2`] = `
"{
    \\"source\\": \\"./src\\",
    \\"destination\\": \\"./docs\\",
    \\"plugins\\": [
        {
            \\"name\\": \\"esdoc-standard-plugin\\"
        },
        {
            \\"name\\": \\"esdoc-ecmascript-proposal-plugin\\",
            \\"option\\": {
                \\"all\\": true
            }
        },
        {
            \\"name\\": \\"esdoc-jsx-plugin\\",
            \\"options\\": {
                \\"enable\\": true
            }
        }
    ]
}
"
`;

exports[`"Add" commands add-esdoc 1`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": []
}
"
`;

exports[`"Add" commands add-esdoc 2`] = `
"{
    name: tomo-test,
    children: [
        {
            name: esdoc.conf.json,
            size: 286,
            type: file
        },
        {
            name: package.json,
            size: 419,
            type: file
        }
    ],
    size: 705,
    type: directory
}
"
`;

exports[`"Add" commands add-esdoc 3`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"scripts\\": {
        \\"lint:docs\\": \\"eslint . --no-eslintrc --rule valid-jsdoc:error --parser babel-eslint\\",
        \\"predocs\\": \\"npm run lint:docs\\",
        \\"docs\\": \\"esdoc -c esdoc.conf.json\\",
        \\"postdocs\\": \\"open-cli ./docs/index.html\\"
    }
}
"
`;

exports[`"Add" commands add-esdoc 4`] = `
"{
    \\"source\\": \\"./src\\",
    \\"destination\\": \\"./docs\\",
    \\"plugins\\": [
        {
            \\"name\\": \\"esdoc-standard-plugin\\"
        },
        {
            \\"name\\": \\"esdoc-ecmascript-proposal-plugin\\",
            \\"option\\": {
                \\"all\\": true
            }
        }
    ]
}
"
`;

exports[`"Add" commands add-eslint (with React) 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .eslintignore,
            size: 36,
            type: file
        },
        {
            name: .eslintrc.js,
            size: 432,
            type: file
        }
    ],
    size: 468,
    type: directory
}
"
`;

exports[`"Add" commands add-eslint (with React) 2`] = `
"module.exports = {
    env: {
        es6: true,
        jest: true
    },
    extends: [
        'omaha-prime-grade',
        'plugin:react/recommended',
        'plugin:jsx-a11y/recommended'
    ],
    parser: 'babel-eslint',
    parserOptions: {
        ecmaFeatures: {
            jsx: true
        }
    },
    plugins: [
        'jsx-a11y'
    ],
    settings: {
        react: {
            version: '16.2'
        }
    }
};"
`;

exports[`"Add" commands add-eslint 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: .eslintignore,
            size: 36,
            type: file
        },
        {
            name: .eslintrc.js,
            size: 154,
            type: file
        }
    ],
    size: 190,
    type: directory
}
"
`;

exports[`"Add" commands add-eslint 2`] = `
"module.exports = {
    env: {
        es6: true,
        jest: true
    },
    extends: [
        'omaha-prime-grade'
    ],
    parser: 'babel-eslint'
};"
`;

exports[`"Add" commands add-jest --browser 1`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": []
}
"
`;

exports[`"Add" commands add-jest --browser 2`] = `
"{
    name: tomo-test,
    children: [
        {
            name: package.json,
            size: 168,
            type: file
        }
    ],
    size: 168,
    type: directory
}
"
`;

exports[`"Add" commands add-jest --browser 3`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"jest\\": {
        \\"testMatch\\": [
            \\"**/__tests__/**/*.(e2e|test).[jt]s?(x)\\"
        ],
        \\"setupFilesAfterEnv\\": [
            \\"<rootDir>/__tests__/setup.js\\"
        ],
        \\"watchPlugins\\": [
            \\"jest-watch-typeahead/filename\\",
            \\"jest-watch-typeahead/testname\\"
        ]
    },
    \\"scripts\\": {
        \\"test\\": \\"jest .*.test.js --coverage\\",
        \\"test:ing\\": \\"npm test -- --watchAll\\"
    }
}
"
`;

exports[`"Add" commands add-jest --browser 4`] = `
"{
    name: tomo-test,
    children: [
        {
            name: __tests__,
            children: [
                {
                    name: example.test.js,
                    size: 337,
                    type: file
                },
                {
                    name: setup.js,
                    size: 370,
                    type: file
                }
            ],
            size: 707,
            type: directory
        },
        {
            name: package.json,
            size: 602,
            type: file
        }
    ],
    size: 1309,
    type: directory
}
"
`;

exports[`"Add" commands add-jest 1`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": []
}
"
`;

exports[`"Add" commands add-jest 2`] = `
"{
    name: tomo-test,
    children: [
        {
            name: package.json,
            size: 168,
            type: file
        }
    ],
    size: 168,
    type: directory
}
"
`;

exports[`"Add" commands add-jest 3`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"jest\\": {
        \\"testMatch\\": [
            \\"**/__tests__/**/*.(e2e|test).[jt]s?(x)\\"
        ],
        \\"watchPlugins\\": [
            \\"jest-watch-typeahead/filename\\",
            \\"jest-watch-typeahead/testname\\"
        ]
    },
    \\"scripts\\": {
        \\"test\\": \\"jest .*.test.js --coverage\\",
        \\"test:ing\\": \\"npm test -- --watchAll\\"
    }
}
"
`;

exports[`"Add" commands add-jest 4`] = `
"{
    name: tomo-test,
    children: [
        {
            name: __tests__,
            children: [
                {
                    name: example.test.js,
                    size: 337,
                    type: file
                }
            ],
            size: 337,
            type: directory
        },
        {
            name: package.json,
            size: 516,
            type: file
        }
    ],
    size: 853,
    type: directory
}
"
`;

exports[`"Add" commands add-marionette 1`] = `
"{
    name: tomo-test,
    children: [
        {
            name: assets,
            children: [
                {
                    name: css,
                    children: [
                        {
                            name: fonts.css,
                            size: 558,
                            type: file
                        },
                        {
                            name: style.css,
                            size: 1094,
                            type: file
                        }
                    ],
                    size: 1652,
                    type: directory
                },
                {
                    name: fonts,
                    children: [
                        {
                            name: SansForgetica-Regular.eot,
                            size: 53094,
                            type: file
                        },
                        {
                            name: SansForgetica-Regular.svg,
                            size: 167134,
                            type: file
                        },
                        {
                            name: SansForgetica-Regular.ttf,
                            size: 52788,
                            type: file
                        },
                        {
                            name: SansForgetica-Regular.woff,
                            size: 25732,
                            type: file
                        },
                        {
                            name: SansForgetica-Regular.woff2,
                            size: 18112,
                            type: file
                        }
                    ],
                    size: 316860,
                    type: directory
                },
                {
                    name: images,
                    children: [
                        {
                            name: blank_canvas.png,
                            size: 88698,
                            type: file
                        },
                        {
                            name: preferences.png,
                            size: 81379,
                            type: file
                        }
                    ],
                    size: 170077,
                    type: directory
                },
                {
                    name: index.html,
                    size: 1268,
                    type: file
                },
                {
                    name: library,
                    children: [
                        {
                            name: .gitkeep,
                            size: 0,
                            type: file
                        }
                    ],
                    size: 0,
                    type: directory
                },
                {
                    name: workers,
                    children: [
                        {
                            name: .gitkeep,
                            size: 0,
                            type: file
                        }
                    ],
                    size: 0,
                    type: directory
                }
            ],
            size: 489857,
            type: directory
        },
        {
            name: src,
            children: [
                {
                    name: components,
                    children: [
                        {
                            name: app.js,
                            size: 1516,
                            type: file
                        },
                        {
                            name: body.js,
                            size: 186,
                            type: file
                        },
                        {
                            name: footer.js,
                            size: 657,
                            type: file
                        },
                        {
                            name: header.js,
                            size: 117,
                            type: file
                        }
                    ],
                    size: 2476,
                    type: directory
                },
                {
                    name: main.js,
                    size: 199,
                    type: file
                },
                {
                    name: plugins,
                    children: [
                        {
                            name: mn.radio.logging.js,
                            size: 3980,
                            type: file
                        },
                        {
                            name: mn.redux.state.js,
                            size: 2016,
                            type: file
                        }
                    ],
                    size: 5996,
                    type: directory
                },
                {
                    name: shims,
                    children: [
                        {
                            name: mn.renderer.shim.js,
                            size: 851,
                            type: file
                        }
                    ],
                    size: 851,
                    type: directory
                }
            ],
            size: 9522,
            type: directory
        }
    ],
    size: 499379,
    type: directory
}
"
`;

exports[`Add Reason with Webpack 1`] = `
"{
    \\"name\\": \\"my-project\\",
    \\"version\\": \\"0.0.0\\",
    \\"description\\": \\"A super cool app/server/tool/library/widget/thingy\\",
    \\"license\\": \\"MIT\\",
    \\"keywords\\": [],
    \\"scripts\\": {
        \\"build:reason\\": \\"bsb -make-world -clean-world\\",
        \\"watch:reason\\": \\"npm run build:reason -- -w\\"
    }
}
"
`;

exports[`Add Reason with Webpack 2`] = `
"{
    \\"bs-dependencies\\": [
        \\"reason-react\\"
    ],
    \\"bsc-flags\\": [
        \\"-bs-super-errors\\"
    ],
    \\"namespace\\": true,
    \\"package-specs\\": [
        {
            \\"module\\": \\"es6\\",
            \\"in-source\\": true
        }
    ],
    \\"ppx-flags\\": [],
    \\"reason\\": {
        \\"react-jsx\\": 3
    },
    \\"refmt\\": 3,
    \\"sources\\": [
        {
            \\"dir\\": \\"src\\",
            \\"subdirs\\": true
        }
    ],
    \\"suffix\\": \\".bs.js\\",
    \\"name\\": \\"my-project\\"
}
"
`;

exports[`Add Reason with Webpack 3`] = `
"{
    name: tomo-test,
    children: [
        {
            name: bsconfig.json,
            size: 474,
            type: file
        },
        {
            name: package.json,
            size: 301,
            type: file
        },
        {
            name: src,
            children: [
                {
                    name: components,
                    children: [
                        {
                            name: App.re,
                            size: 116,
                            type: file
                        },
                        {
                            name: Example.re,
                            size: 126,
                            type: file
                        }
                    ],
                    size: 242,
                    type: directory
                }
            ],
            size: 242,
            type: directory
        }
    ],
    size: 1017,
    type: directory
}
"
`;
